components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java

```diff
@@ -705,6 +705,17 @@ public void testHandleAuthenticationFailureWithException() throws Exception {
         assertNull(oAuth2Service.handleAuthenticationFailure(oAuth2Parameters));
     }
 
+    @Test
+    public void testValidateClientInfoWIthQueryParamsInCallbackURL() throws Exception {
+
+        String clientId = UUID.randomUUID().toString();
+        getOAuthAppDO(clientId, "dummyGrantType", "https://wso2.com?dummy1=1&dummy=2",
+                "carbon.super");
+        OAuth2ClientValidationResponseDTO oAuth2ClientValidationResponseDTO = oAuth2Service.
+                validateClientInfo(clientId, "https://wso2.com\\?dummy1=1&dummy=2");
```

Comment:
Shouldn't we pass this without escapting here?

Reply:
The one comes in the auth request is similar to this